### ISSUE TITLE
Call to restart plugin function trashes plugin configuration file

### ISSUE THREAD

**Issue by preeve9534** [SignalK/signalk-server#1641]
I can hardly believe that I am reporting this issue, but having assumed that I had an error in my code and applied due diligence, here we go...

I have a plugin that under some exceptional circumstance calls the restart plugin function passed as the second argument to ```plugin.start```. My start function is declared as ```plugin.start = function(options, restartPlugin) {```

Whilst a call to ```restartPlugin()``` completes successfully the plugin restarts with no configuration options and on closer inspection the reason is that the 'configuration' property has been deleted from the plugin configuration file, leaving just:
```
{
  "enabled": true,
  "enableDebug": "true"
}
```
(This is how these properties were set previously)

If I leave all other program logic intact and comment out the call to ```restartPlugin()``` my configuration file remains untouched. Comment in the call to the restart plugin function and my configuration file is trashed.

Question is: why is the restart plugin function touching my configuration file?

**Comment by tkurki** [comment]
Oh, the docs are lacking, you are supposed to pass in a new configuration object. So `restart` is actually `save this new configuration and restart`.

**Comment by preeve9534** [comment]
Thanks @tkurki. I guessed some such. However...

I have some questions on the logic of this though:

1. Mostly I simply want to restart a plugin and NOT change its configuration, so having to pass in an argument seems daft.
2. A more sensible default behaviour would surely be to preserve the existing configuration if no argument is passed?
3. Just exactly how does a call to ```restart``` differ from a call to ```app.savePluginOptions```? 
4. With #1559 still unfixed, I'm guessing that ```restart``` with an argument is similarly problematic in one of its primary use-cases.

**Comment by tkurki** [comment]
To be honest it has been such a long time since this has been added I can't remember the exact details / use case for this. And since the docs are off I don't think people have been using this much if at all.

1. I don't disagree, but this is how it has been working
2. I don't disagree
3. Is it me being daft, but isn't it pretty obvious: savePluginOptions does what it says on the tin, restart saves & restarts, as I wrote above
